tests/unittests/test_request_timeouts.py

```diff
@@ -0,0 +1,134 @@
+        """
+            Verify that if request_timeout is provided in config(integer value) then it should be use
+        """
+        # Initialize MailchimpClient object
+        client = MailchimpClient({'access_token': 'as', "request_timeout": REQUEST_TIMEOUT_INT}) # integer timeout in config
+        
+        # Call request method which call requests.Session.request with timeout
+        client.request('GET', "http://test", "base_url")
+        
+        # Verify requests.Session.request is called with expected timeout
+        args, kwargs = mocked_request.call_args
+        self.assertEqual(kwargs.get('timeout'), REQUEST_TIMEOUT_FLOAT) # Verify timeout argument
+
+    def test_float_request_timeout_in_config(self, mocked_request):
+        """
+            Verify that if request_timeout is provided in config(float value) then it should be use
+        """
+        # Initialize MailchimpClient object
+        client = MailchimpClient({'access_token': 'as', "request_timeout": REQUEST_TIMEOUT_FLOAT}) # float timeout in config
+        
+        # Call request method which call requests.Session.request with timeout
+        client.request('GET', "http://test", "base_url")
+        
+        # Verify requests.Session.request is called with expected timeout
+        args, kwargs = mocked_request.call_args
+        self.assertEqual(kwargs.get('timeout'), REQUEST_TIMEOUT_FLOAT) # Verify timeout argument
+
+    def test_string_request_timeout_in_config(self, mocked_request):
+        """
+            Verify that if request_timeout is provided in config(string value) then it should be use
+        """
+        # Initialize MailchimpClient object
+        client = MailchimpClient({'access_token': 'as', "request_timeout": REQUEST_TIMEOUT_STR}) # string timeout in config
+        
+        # Call request method which call requests.Session.request with timeout
+        client.request('GET', "http://test", "base_url")
+        
+        # Verify requests.Session.request is called with expected timeout
+        args, kwargs = mocked_request.call_args
+        self.assertEqual(kwargs.get('timeout'), REQUEST_TIMEOUT_FLOAT) # Verify timeout argument
+
+    def test_empty_string_request_timeout_in_config(self, mocked_request):
+        """
+            Verify that if request_timeout is provided in config with empty string then default value is used
+        """
+        # Initialize MailchimpClient object
+        client = MailchimpClient({'access_token': 'as', "request_timeout": ""}) # empty string timeout in config
+        
+        # Call request method which call requests.Session.request with timeout
+        client.request('GET', "http://test", "base_url")
+        
+        # Verify requests.Session.request is called with expected timeout
+        args, kwargs = mocked_request.call_args
+        self.assertEqual(kwargs.get('timeout'), REQUEST_TIMEOUT_INT) # Verify timeout argument
```

Comment:
@prijendev Why int value is expected in this test case?

Reply:
If the request_timeout value is not passed in the config then the default value that is int value(300) is used.